### DISCUSSION TITLE
Experimental Features for Haystack

### DISCUSSION THREAD

**Discussion by TuanaCelik** [deepset-ai/haystack-experimental#80] [Announcements]
Hi dear Haystack community 👋

Recently, we announced the new `haystack-experimental` package which gets auto installed when you `pip install haystack-ai`.
This package includes a bunch of experimental features we are trying out, and you can use them by importing a given feature `from haystack_experimenta`, for example:

```
from haystack_experimental.components.generators import FoobarGenerator
```

We ask you to please check out these features, and let us know what you think in their corresponding Discussion Boards.

**🙏 Please do not open your own discussions about Haystack here. Use the [Haystack repository](https://github.com/deepset-ai/haystack) for any non-experimental related issues/discussions.**

**Comment by lacebal** [comment]
Hi, 

Can you please clarify the release dates of the experimental features, from the doc,

> Each experimental feature has a default lifespan of 3 months starting from the date of the first non-pre-release build that includes it. Once it reaches the end of its lifespan, the experiment will be either:
> 
> - Merged into Haystack core and published in the next minor release, or
> - Released as a Core Integration, or
> - Dropped.

But all the experiments are already past their end date. For example `AsyncPipeline` is 1 year past the end date.

**Reply by bilgeyucel** [reply]
Hi @lacebal,  
The usual lifespan of an experiment is 3 months. However, sometimes, we need more time to test the experiment and gather feedback. The experiment list will be updated in the upcoming days! 🙌 

Regarding `AsyncPipeline`, I believe there is a mistake. The date is supposed to be "February 2025" and the `AsyncPipeline` experiment has already been merged into `haystack` with the [2.10 release](https://github.com/deepset-ai/haystack/releases/tag/v2.10.0) so you can use it from the core package! 😊
